src/library.js

```diff
@@ -1714,25 +1714,25 @@ LibraryManager.library = {
   //    being compiled. Not sure how to tell LLVM to not do so.
   // ==========================================================================
 
+#if MAIN_MODULE == 0
+  dlopen: function(/* ... */) {
+    abort("To use dlopen, you need to use Emscripten's linking support, see https://github.com/kripken/emscripten/wiki/Linking");
+  },
+  dlclose: 'dlopen',
+  dlsym:   'dlopen',
+  dlerror: 'dlopen',
+  dladdr:  'dlopen',
+#else
```

Comment:
`// MAIN_MODULE == 0` on the else and the endif, please

Reply:
ok, I've added `// MAIN_MODULE != 0` after the `#else` (note the negation). For `#endif` it already was

```
#endif // MAIN_MODULE != 0
```

I put it that way because endif is very far from if/else, and by just looking at the endif it is not clear whether it was only if/endif, or it was if/else/endif, and this way the condition comment on the endif indicates the condition under which `... endif` block is actually covered. For else it follows the same logic - it show the condition under which code block covered by else is activated.

We aleady have several cases where such approach is used, e.g. here:

https://github.com/kripken/emscripten/blob/1.38.20-11-gfc1844db2/src/library.js#L526-L539
https://github.com/kripken/emscripten/blob/1.38.20-11-gfc1844db2/src/library_html5.js#L2179-L2238

I hope it is ok.